src/dswx_sar/dswx_comparison.py

```diff
@@ -0,0 +1,208 @@
+
+       Parameters
+       ----------
+       image_1 : numpy.ndarray
+            First input image
+       image_2: numpy.ndarray
+            Second input image
+       prefix: str
+            Prefix to the message printed to the user
+    """
+    flag_error_found = False
+    for i in range(image_1.shape[0]):
+        for j in range(image_1.shape[1]):
+            if (abs(image_1[i, j] - image_2[i, j]) <=
+                    COMPARE_DSWX_SAR_PRODUCTS_ERROR_TOLERANCE):
+                continue
+            print(prefix + f'     * input 1 has value'
+                  f' "{image_1[i, j]}" in position'
+                  f' (x: {j}, y: {i})'
+                  f' whereas input 2 has value "{image_2[i, j]}"'
+                  ' in the same position.')
+            flag_error_found = True
+            break
+        if flag_error_found:
+            break
+
+def _compare_dswx_sar_metadata(metadata_1, metadata_2):
+    """
+    Compare DSWx-SAR products' metadata
+
+       Parameters
+       ----------
+       metadata_1 : dict
+            Metadata of the first DSWx-SAR product
+       metadata_2: dict
+            Metadata of the second
+    """
+    metadata_error_message = None
+    flag_same_metadata = len(metadata_1.keys()) == len(metadata_2.keys())
+    if not flag_same_metadata:
+        metadata_error_message = (
+            f'* input 1 metadata has {len(metadata_1.keys())} entries'
+            f' whereas input 2 metadata has {len(metadata_2.keys())} entries.')
+        
+        set_1_m_2 = set(metadata_1.keys()) - set(metadata_2.keys())
+        if len(set_1_m_2) > 0:
+            metadata_error_message += (' Input 1 metadata has extra entries'
+                                       ' with keys:'
+                                       f' {", ".join(set_1_m_2)}.')
+        set_2_m_1 = set(metadata_2.keys()) - set(metadata_1.keys())
+        if len(set_2_m_1) > 0:
+            metadata_error_message += (' Input 2 metadata has extra entries'
+                                       ' with keys:'
+                                       f' {", ".join(set_2_m_1)}.')
+    else:
+        for k1, v1, in metadata_1.items():
+            if k1 not in metadata_2.keys():
+                flag_same_metadata = False
+                metadata_error_message = (
+                    f'* the metadata key {k1} is present in'
+                    ' but it is not present in input 2')
+                break
+            # Exclude metadata fields that are not required to be the same
+            if k1 in COMPARISON_EXCEPTION_LIST:
+                continue
+            if metadata_2[k1] != v1:
+                flag_same_metadata = False
+                metadata_error_message = (
+                    f'* contents of metadata key {k1} from'
+                    f' input 1 has value "{v1}" whereas the same key in'
+                    f' input 2 metadata has value "{metadata_2[k1]}"')
+                break
```

Comment:
```suggestion
        return metadata_error_message, flag_same_metadata

    for k1, v1, in metadata_1.items():
        if k1 not in metadata_2.keys():
            flag_same_metadata = False
            metadata_error_message = (
                f'* the metadata key {k1} is present in'
                ' but it is not present in input 2')
            break
        # Exclude metadata fields that are not required to be the same
        if k1 in COMPARISON_EXCEPTION_LIST:
            continue
        if metadata_2[k1] != v1:
            flag_same_metadata = False
            metadata_error_message = (
                f'* contents of metadata key {k1} from'
                f' input 1 has value "{v1}" whereas the same key in'
                f' input 2 metadata has value "{metadata_2[k1]}"')
            break
```

Here is the suggestion about how the early return can be exercised.